aptos-move/framework/src/natives/account.rs

```diff
@@ -16,29 +16,36 @@ use move_deps::{
 use smallvec::smallvec;
 use std::collections::VecDeque;
 
-pub fn native_create_signer(
-    context: &mut NativeContext,
+pub fn native_create_address(
+    _context: &mut NativeContext,
     ty_args: Vec<Type>,
     mut arguments: VecDeque<Value>,
 ) -> PartialVMResult<NativeResult> {
     debug_assert!(ty_args.is_empty());
     debug_assert!(arguments.len() == 1);
 
-    let address = pop_arg!(arguments, AccountAddress);
-    let cost = native_gas(context.cost_table(), NativeCostIndex::CREATE_SIGNER, 0);
-    Ok(NativeResult::ok(cost, smallvec![Value::signer(address)]))
+    let cost = GasCost::new(super::cost::APTOS_CREATE_ADDRESS, 1).total();
+    let bytes = pop_arg!(arguments, Vec<u8>);
+    let address = AccountAddress::from_bytes(bytes);
+    if let Ok(address) = address {
+        Ok(NativeResult::ok(cost, smallvec![Value::address(address)]))
+    } else {
+        Ok(NativeResult::err(
+            cost,
+            super::status::NFE_UNABLE_TO_PARSE_ADDRESS,
+        ))
+    }
 }
 
-/// NOTE: this function will be deprecated after the Aptos v3 release, but must
-/// remain for replaying old transactions
-pub fn native_destroy_signer(
+pub fn native_create_signer(
```

Comment:
Code diffing in Github is so dumb. This native_create_signer looks exactly the same as before (which was earlier in the file) but it's being shown as a diff.